### PR TITLE
Better support for custom user models

### PR DESCRIPTION
**Pull request by joshkel**
This adds support for `User.get_username` and non-default usernames. I also fixed the Tox configuration.

In order to keep both Django 1.4 support and 100% test coverage, the test code is kind of a hack. I'm open to suggestions on ways to improve that.

### CODE REVIEW

**Comment by mattrobenolt** [comment]
Hey @joshkel, thanks for the patch!

Can you fix the tests please?

**Comment by joshkel** [comment]
I'm sorry. I checked tox and the unit tests but overlooked inv test and flake8. It should be fixed now.

File: tox.ini [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
-envlist = py26, py27, py32, py33, py34, pypy
+envlist = py26-django14, py27, py32, py33, py34, pypy
 
 [testenv]
-deps = pytest
+deps =
+       django14: django >=1.4.2,<1.5
+       pytest
        pytest-cov
-       pytest-django-lite
```
Comment (mattrobenolt) [inline tox.ini]: Good catch! :)

File: tests/models.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import django
+from django.db import models
+try:
+    from django.contrib.auth.models import AbstractBaseUser
+except ImportError:
+    # Django 1.4
+    from django.contrib.auth.models import User as AbstractBaseUser
+
+
+class Django14User(AbstractBaseUser):
```
Comment (mattrobenolt) [inline tests/models.py]: From my understanding, you should be able to use https://pytest-django.readthedocs.org/en/latest/helpers.html#django-user-model, correct?

**Comment by mattrobenolt** [comment]
Hey @joshkel, I apologize for the long delay. I lost this in my inbox. :(

I had one comment regarding the `Django14User` model trickery, and if you can address that, it'd be much appreciated. If not, I can take over and swap it out.

**Comment by joshkel** [comment]
Regarding the delay, no problem; I'm sorry I didn't reply sooner.

Thanks for the link to pytest and `django_user_model`. Maybe I'm missing something, but I don't see how it would suffice by itself. If I understand correctly, the purpose of `django_user_model` is to handle the difference between `User` on Django 1.4 (because 1.4 doesn't have built-in support for replacing the user model) and `get_user_model()` on 1.5 and above. Instead, the test needs to substitute a custom user model for a single test.

It looks like I could use [TestCase.settings()](https://docs.djangoproject.com/en/1.4/topics/testing/#overriding-settings) plus `django_user_model`, but after looking at this more, I think it's probably easiest to delete some of the hackery and just add a "pragma: nocover" to not complain about the Django 1.4 support. Please see my updated PR.

If I'm missing anything, or if there's anything you'd like me to change, please let me know.

**Comment by mattrobenolt** [comment]
Awesome, yeah, that makes sense.

Thanks again! :cake:

**Comment by joshkel** [comment]
Thanks!

**Pull request merged**